src/rust/rust-proto-new/src/graplinc/grapl/api/plugin_registry/v1beta1_server.rs

```diff
@@ -94,9 +96,43 @@ where
 {
     async fn create_plugin(
         &self,
-        request: Request<proto::CreatePluginRequest>,
+        request: Request<tonic::Streaming<proto::CreatePluginRequest>>,
     ) -> Result<Response<proto::CreatePluginResponse>, tonic::Status> {
-        execute_rpc!(self, request, create_plugin)
+        let mut proto_request = request.into_inner();
+
+        // Spin up two Futures
+        // - one converting incoming protobuf requests to Rust-native requests
+        // - one calling the `.create_plugin` handler
+        // (with a tx/rx communicating between the two threads)
+
+        let (mut tx, rx) = futures::channel::mpsc::channel(8);
+
+        let proto_to_native_thread = async move {
+            ({
+                while let Some(req) = proto_request.next().await {
+                    let req = req?.try_into()?;
+                    tx.send(req)
+                        .await
+                        .map_err(|e| Status::internal(e.to_string()))?;
+                }
+                Ok(())
+            } as Result<(), Status>)
+        };
+
+        let api_handler_thread = async move {
+            ({ self.api_server.create_plugin(rx).await.map_err(Into::into) }
```

Comment:
the trait bound `protocol::status::Status: From<<T as PluginRegistryApi>::Error>` is not satisfied
required because of the requirements on the impl of `FromResidual<Result<Infallible, <T as PluginRegistryApi>::Error>>` for